### DISCUSSION TITLE
Use autoincrement when using sesssion.add

### DISCUSSION THREAD

**Discussion by caynebrister1** [sqlalchemy/sqlalchemy#10214] [Usage Questions] [answered]
I am trying to use session.add() to add an entry to a table, but I get the error "missing 1 required positional argument: 'id'". Is there a way I can add an entry without specifying the ID that way sqlalchemy can choose the incremented ID itself. The reason is because this table is changing constantly and I worry that just getting the highest ID and incrementing it myself could cause issues if there is enough of a delay between retrieving the highest ID, incrementing it, then adding the entry.

**Comment by zzzeek** [comment]
session.add() does not accept an "id" argument so I dont know to what this refers.  would need to see stack trace plus code example.

**Reply by zzzeek** [reply, marked as answer]
that said, all of the examples in SQLAlchemy's tutorials use auto-incrementing ids, see https://docs.sqlalchemy.org/en/20/orm/quickstart.html .  if you are using dataclasses, set init=False on entries you dont want in your constructor.

**Reply by caynebrister1** [reply]
Sorry, I didn't provide this originally, thanks ahead of time!
Here is the func I am working on

```
def add_to_submission_table(repair):
    session = Session()
    repair_entry = RepairsToSubmit(**repair)
    session.add(repair_entry)
    session.commit()
    session.close()
```

Here is the file dedicated to defining the DB models:

```
from sqlalchemy import create_engine, insert, ForeignKey, Column, Integer
from sqlalchemy import exc, String, CHAR, Table, MetaData, select
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker

Base = declarative_base()
engine = create_engine('sqlite:///data/db.sqlite?check_same_thread=False')

class RepairsToSubmit(Base):
    __tablename__ = 'reapirs_to_submit'

    id = Column(Integer, primary_key=True)
    technician = Column('technician', String)
    service = Column('service', String)
    nisd_num = Column('nisd_num', String)
    campus = Column('campus', String)
    device_model = Column('device_model', String)
    parts = Column('parts', String)
    damage_type = Column('damage_type', String)
    notes = Column('notes', String)

    def __init__(self, id, technician, service, nisd_num, campus, device_model, parts, damage_type, notes):
        self.id = id
        self.technician = technician
        self.service = service
        self.nisd_num = nisd_num
        self.campus = campus
        self.device_model = device_model
        self.parts = parts
        self.damage_type = damage_type
        self.notes = notes
    
    def __repr__(self):
        return "<Repair('%s')>" % (self.nisd_num)

Base.metadata.create_all(bind=engine)
Session = sessionmaker(bind=engine)
```

**Reply by CaselIT** [reply]
as replied in the chat, you need to modify your init to make id optional. This is a general python question, has no connection with sqlalchemy

**Reply by caynebrister1** [reply]
Sorry, I hadn't seen your reply, that looks like the issue.
